### PR TITLE
Use `noexcept` more.

### PR DESCRIPTION
**Pull request by waywardmonkeys**
This applies it to swap functions as well as move constructors, move assignment, etc.

### CODE REVIEW

File: src/ast/recfun_decl_plugin.cpp [hunk 1]
```diff
@@ -606,7 +606,7 @@ namespace recfun {
         : m_lhs(from.m_lhs),
           m_def(from.m_def),
           m_args(from.m_args) {}
-    case_expansion::case_expansion(case_expansion && from)
+    case_expansion::case_expansion(case_expansion && from) noexcept
```
Comment (nunoplopes) [inline src/ast/recfun_decl_plugin.cpp:609]: these copy/move constructors should just be deleted

Comment (waywardmonkeys) [reply src/ast/recfun_decl_plugin.cpp:609]: @nunoplopes Just this one? Or more?

Comment (nunoplopes) [reply src/ast/recfun_decl_plugin.cpp:609]: I only spotted this one, but there might be others. The ones that do reference counting can't go; everything else should go.

Comment (NikolajBjorner) [reply src/ast/recfun_decl_plugin.cpp:609]: Note that case-expansion is not on a hot path so seems overtuned to add a move constructor here in the first place.

Comment (waywardmonkeys) [reply src/ast/recfun_decl_plugin.cpp]: In the same file, there was a move constructor for `body_expansion` as well (which was already `noexcept`), so I removed that one as well.

File: src/ast/recfun_decl_plugin.h [hunk 2]
```diff
@@ -302,7 +302,6 @@ namespace recfun {
         expr_ref_vector     m_args;
         case_expansion(recfun::util& u, app * n);
         case_expansion(case_expansion const & from);
```
Comment (nunoplopes) [inline src/ast/recfun_decl_plugin.h:304]: the copy constructor must go too. You can't have just one of them. Either all or nothing.

File: src/ast/recfun_decl_plugin.h [hunk 3]
```diff
@@ -325,8 +324,6 @@ namespace recfun {
             m_pred(pred), m_cdef(&d), m_args(args) {}
         body_expansion(body_expansion const & from):
```
Comment (nunoplopes) [inline src/ast/recfun_decl_plugin.h:325]: idem

File: src/util/f2n.h [hunk 4]
```diff
@@ -45,7 +45,7 @@ class f2n {
         m_manager.set(m_one, ebits, sbits, 1);
     }
 
-    f2n(f2n && other) : m_manager(other.m_manager), m_mode(other.m_mode), m_ebits(other.m_ebits), m_sbits(other.m_sbits),
+    f2n(f2n && other) noexcept : m_manager(other.m_manager), m_mode(other.m_mode), m_ebits(other.m_ebits), m_sbits(other.m_sbits),
```
Comment (nunoplopes) [inline src/util/f2n.h:48]: this one can go too.

**Comment by waywardmonkeys** [comment]
Updated, rebased to fix a merge conflict, and force pushed.

File: src/util/f2n.h (hunk 4 above)
Comment (waywardmonkeys) [reply src/util/f2n.h:48]: When this one goes away, there's a build error on Ubuntu in CI. I've added it back for now.

**Pull request merged**